src/GraphQL.Client/GraphQLHttpClient.cs

```diff
@@ -111,6 +116,62 @@ public IObservable<GraphQLResponse<TResponse>> CreateSubscriptionStream<TRespons
 
     #region Private Methods
 
+    private async Task<GraphQLHttpResponse<TResponse>> SendAPQHttpRequestAsync<TResponse>(GraphQLRequest request, CancellationToken cancellationToken = default)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        var savedQuery = request.Query;
+        bool useAPQ = false;
+
+        if (request.Query != null && _useAPQ && Options.EnableAutomaticPersistedQueries(request))
+        {
+            // https://www.apollographql.com/docs/react/api/link/persisted-queries/
+            const int APQ_SUPPORTED_VERSION = 1;
+            useAPQ = true;
+            // TODO: I suggest to change request.Extension type to public Dictionary<string, object?>?
+            //request.Extensions ??= new();
+            //request.Extensions["persistedQuery"] = new Dictionary<string, object>
+            //{
+            //    ["version"] = APQ_SUPPORTED_VERSION,
+            //    ["sha256Hash"] = Hash.Compute(request.Query),
+            //};
+            request.Extensions ??= new
+            {
```

Comment:
doesn't it need those values to be wrapped within a 'PersistedQuery' property?

Reply:
yep, forgot about 'PersistedQuery' when copy-pasted code. Operation name is out of scope here, APQ has nothing to do with operatio name.